### ISSUE TITLE
Unexpected Dialyzer issue

### ISSUE THREAD

**Issue by mathieurousseau** [dcrck/ultraviolet#1]
Elixir 17. latest version of ultraviolet.

Following code is giving me a dialyzer issue:
```elixir
defmodule MyModule do
  require Logger

  def testing do
    sport_color = "pink"

    case Ultraviolet.new(sport_color) do
      {:error, m} -> Logger.error("should_not_happen", reason: m)
      {:ok, color} -> color
    end
  end
end
```

error : 
```elixir
lib/my_module.ex:9:pattern_match
The pattern can never match the type.

Pattern:
{:ok, _color}

Type:
{:error, _}

________________________________________________________________________________
done (warnings were emitted)
Halting VM with exit status 2
```

<img width="1296" alt="image" src="https://github.com/user-attachments/assets/2b9ec9d2-1603-4be8-9253-fc886d14c5d6" />

**Comment by mathieurousseau** [comment]
Created a simple repo:

https://github.com/mathieurousseau/ultraviolet_dialyzer

**Comment by dcrck** [comment]
Thanks for reporting this.

I think the issue is due to how Ultraviolet handles the W3CX11 colors, like `"pink"`. Basically, I open an external file and create `new` declarations for each color on compilation. See https://github.com/dcrck/ultraviolet/blob/eb6f07977c407d9926f9449473528d4654d2f87f/lib/ultraviolet/color.ex#L73-L88 and https://github.com/dcrck/ultraviolet/blob/main/lib/ultraviolet/named-colors.txt

I don't really know how to fix this at the moment, but I'll ask around. I would say just ignore that dialyzer error for now.

**Comment by dimitarvp** [comment]
This will likely not help you at all but here's what I have on a freshly cloned project (Elixir 1.18.1-OTP-27, Erlang 27.2):

```bash
❯ mix do deps.get, compile
Resolving Hex dependencies...
Resolution completed in 0.009s
Unchanged:
  decimal 2.3.0
  dialyxir 1.4.5
  erlex 0.2.7
  ultraviolet 0.1.0
* Getting ultraviolet (Hex package)
* Getting dialyxir (Hex package)
* Getting erlex (Hex package)
* Getting decimal (Hex package)
==> decimal
Compiling 4 files (.ex)
Generated decimal app
==> ultraviolet
Compiling 14 files (.ex)
     warning: the following clause will never match:

         {:ok, colors}

     because it attempts to match on the result of:

         Ultraviolet.ColorBrewer.colors(palette, count)

     which has type:

         dynamic({:error, :not_found})

     typing violation found at:
     │
 743 │       {:ok, colors} -> Scale.new(colors, options)
     │       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     │
     └─ lib/ultraviolet.ex:743: Ultraviolet.scale/2

Generated ultraviolet app
==> erlex
Compiling 1 file (.xrl)
Compiling 1 file (.yrl)
src/erlex_parser.yrl: Warning: conflicts: 27 shift/reduce, 0 reduce/reduce
Compiling 2 files (.erl)
Compiling 1 file (.ex)
Generated erlex app
==> dialyxir
Compiling 67 files (.ex)
Generated dialyxir app
==> ultraviolet_dialyzer
Compiling 2 files (.ex)
Generated ultraviolet_dialyzer app
```

Note that this might be Elixir's new type checker being overzealous; 1.18 is the first release where the new type checker is being used in production. If that's the case then it might be worth escalating to the core Elixir team itself.

**Comment by dcrck** [comment]
This should now be fixed in `v0.1.1` :).
